### PR TITLE
QueryField: Make editor navigable with  keyboard

### LINKED ISSUE #35378: Prometheus Query Editor: Obstructs keyboard navigation
**Steps to reproduce**:
1. Go to Explore
2. Choose a Prometheus datasource
3. Navigate to the query field with tab

**Actual Result**:
A tab is put in the field

**Expected Result**
Some option to continue to navigate with tab. Discussions in Slack has come with a couple of suggestions, `esc` would exit edit mode. Toggling tab-trapper behaviour with a global `Ctrl+M`

**Relevant WCAG Criteria:** 2.1.2 WCAG Criterion

### PR DESCRIPTION
**Pull request by tskarhed**
**What this PR does / why we need it**:
Fixes #35378

Use escape key to exit focus and you can tab to change focus unless you've already written something.

### CODE REVIEW

File: packages/grafana-ui/src/components/QueryField/QueryField.tsx [hunk 1]
```diff
@@ -119,6 +124,16 @@ export class QueryField extends React.PureComponent<QueryFieldProps, QueryFieldS
     }
   }
 
+  onKeyDown = (event: KeyboardEvent, editor: CoreEditor, next: Function) => {
```
Comment (kaydelaney) [inline packages/grafana-ui/src/components/QueryField/QueryField.tsx:127]:
```suggestion
  onKeyDown = (keyEvent: Event, editor: CoreEditor, next: Function) => {
    const event = keyEvent as KeyboardEvent;
```
Unfortunately I think you need to do something like this to satisfy typechecks

**Review by kaydelaney — CHANGES_REQUESTED** [review]
Looks good and works! Do you think you could add an `aria-description` attribute informing the user what the component is and that "esc" can be pressed to escape focus?

**Comment by tskarhed** [comment]
I added a tooltip, but it behaves weirdly. It should be shown on focus, but you kinda have to play around with the mouse to make it show. Any idea how to solve this?

**Comment by tskarhed** [comment]
I'm currently trying to get a decent label with the escape key info working. I'm having some issues with the Tooltip, so might go for another solution there.

**Comment by tskarhed** [comment]
Focus styles probably need to be added, and I temporarily used `FormValidationMessage`. This should be changed to something neutral.

**Comment by tskarhed** [comment]
Okay, this looks decent. There is however a problem with the focus/blur if there is some content in the input. Not sure what to do about it.

Removed the label, to get some more UX opinions first.

**Comment by tskarhed** [comment]
Closing this in favor of a future Monaco editor.

**Pull request closed without merging**